cytosnake/tests/functional/test_cli.py

```diff
@@ -1,226 +1,330 @@
+# --------------
+# helper functions
+# --------------
+@dataclass
+class DataFiles:
+    """Structured datatype representation that contains all files in a selected dataset
 
-    def __call__(self) -> None:
-        shutil.rmtree(self.tmp_path)
+    Attributes:
+    -----------
+    metdata: str | list[str]
+        metadata directory name
+    plate_data: list[str]
+        list of plate data (parquet or sqlite files)
+    barcode : str
+        barcode file name
 
+    Returns
+    -------
+    DataFiles
+        DataStructure representation of test dataset files
+    """
 
-def transfer_data(
-    test_dir: pathlib.Path,
-    n_plates: int,
-    n_platemaps: int,
-    metadata_dir_name: Optional[str] = "metadata",
-    testing_data_dir="emptyfiles",
-) -> None:
-    """Wrapper function that transfer datasets found within the pytest module and
-    transfers it to the assigned directory where pytest is conducting the functional
-    tests.
-
-    The test dataset is transfered from the `dataset` directory located in the
-    same directory as the functional test module. The dataset directory contains
-    empty files that emulates inputs that a user will put in to CytoSnake.
+    # required parameters
+    dataset_dir: str | pathlib.Path
+
+    # extracted files
+    metadata: str | list[str] = None
+    plate_data: str | list[str] = None
+    barcode: str = None
+
+    # extracting file paths and setting into dataclass attributes
+    def __post_init__(self):
+        self._extract_content_files()
+
+    def _extract_content_files(self):
+        """extracts all files within given dataset folder and sets the DataFile dataclass
+        attributes
+
+        Raises
+        ------
+        TypeError
+            raised if dataset_dir is not a str or pathlib.Path object.
+            raised if plate data is not parquet or sqlite file
+        """
+        # get all top level files
+        if not isinstance(self.dataset_dir, (str, pathlib.Path)):
+            raise TypeError(
+                "dataset_dir must be a string or pathlib.Path object"
+            )
+        if isinstance(self.dataset_dir, str):
+            self.dataset_dir = pathlib.Path(self.dataset_dir).resolve(
+                strict=True
+            )
+
+        # get all files
+        all_files = list(self.dataset_dir.glob("*"))
+
+        # get data files
+        plate_data = [
+            str(fpath.name)
+            for fpath in all_files
+            if fpath.suffix == ".parquet" or fpath.suffix == ".sqlite"
+        ]
+        self.plate_data = plate_data
+
+        # get metadata_dir
+        meta_data_path = [
+            str(fpath.name) for fpath in all_files if fpath.is_dir()
+        ]
+        self.metadata = (
+            meta_data_path[0] if len(meta_data_path) == 1 else meta_data_path
+        )
+
+        # get barcode
+        barcode_path = [
+            str(fpath.name) for fpath in all_files if fpath.suffix == ".txt"
+        ]
```

Comment:
I don't have as much experience with barcode files yet but wonder if [this](https://github.com/WayScience/Benchmarking_NF1_data/blob/main/CellProfiler_pipelines/Metadata/barcode_platemap.csv) might be an example from another context which helps (showing the file potentially in `.csv` form). Maybe another way of considering this comment would be to think about what value the if-branch in the comprehension adds; if it were removed would you retain the functionality you seek and also simplify things? If it yields an error downstream on removal it could be a good indication of necessary development work.

Reply:
Just spoke with Greg. In order to prevent over engineering, we're going to start narrow (.csv and .txt fies). If there happens to be large diversity of different formats in barcodes, then we would have to think of another approach to validate barcode files in CytoSnake